tasks/main.yml

```diff
@@ -1,11 +1,26 @@
 # SPDX-License-Identifier: MIT
 ---
-# Put the tasks for your role here.
+
+- name: Ensure time source is provided if managing timesync
+  fail:
+    msg: "ad_integration_timesync_source must be provided if manage timesync is true"
```

Comment:
the recommendation is not to quote strings unless they require quoting - this string doesn't require quoting

Reply:
see below also

Reply:
Also - yamllint and ansible-lint may complain about the line length - you can either add suppression comments to the code, or wrap like this:
```yaml
- name: Ensure manage_crypt_policies is set with crypto_allow_rc4
  fail:
    msg: >-
      ad_integration_manage_crypto_policies must be true if
      ad_integration_allow_rc4_crypto is true
  when:
    - ad_integration_allow_rc4_crypto | bool
    - not ad_integration_manage_crypto_policies
```
note the `| bool` was omitted in the last one - the use of the "not" here means it isn't a "bare variable"

Reply:
Fixed.